src/Client/V3/Settings.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lalamove\Client\V3;
+
+use Psr\Log\LoggerInterface;
+
+class Settings
+{
+    public $host;
+    public $apiKey;
+    public $apiSecret;
+    public $country;
+    public $version;
+    public $logger;
+
+    const COUNTRY_HONGKONG = 'HK';
+    const COUNTRY_PHILIPPINES = 'PH';
+    const COUNTRY_SINGAPORE = 'SG';
+    const COUNTRY_THAILAND = 'TH';
+    const COUNTRY_TAIWAN = 'TW';
+
+    const VERSION_3 = 3;
+
+    /**
+     * Pass in either individual settings:
+     *
+     * $settings = new Settings('host.com', '1234567890', // etc...
+     *
+     * ... or an associative array of settings:
+     *
+     * $settings = new Settings(['host' => 'host.com', 'customerId' => '1234567890', 'privateKey' => // etc...
+     *
+     * @param string|array $host
+     * @param string $apiKey
+     * @param string $apiSecret
+     * @param string $country
+     * @param int $version
+     * @param LoggerInterface|null $logger to log each request/response pair the HTTP client handles
+     */
+    public function __construct(
+        $host,
+        $apiKey = '',
+        $apiSecret = '',
+        $country = '',
+        $version = self::VERSION_3,
+        LoggerInterface $logger = null
```

Comment:
Can add types where possible:

```php
    public function __construct(
        $host, // cannot add a type here as a union is needed: string|array
        string $apiKey = '',
        string $apiSecret = '',
        string $country = '',
        string $version = self::VERSION_3,
        LoggerInterface $logger = null
    ) {
        // ...
    }
```

Reply:
(Ignore the comment above).

Let's refactor instead of supporting the union type.

Update the signature:

```
public function __construct(
        string $host,
        string $apiKey = '',
        string $apiSecret = '',
        string $country = '',
        string $version = self::VERSION_3,
        LoggerInterface $logger = null
    ) {
        // ...
    }
```

... and then update the constructor:

```
    {
        $this->host       = $host;
        $this->apiKey = $apiKey;
        $this->apiSecret = $apiSecret;
        $this->country    = $country;
        $this->version    = $version;
        $this->logger     = $logger;
        // note: removed lines 56-60
    }
```